tests/serialization/test_suite_result_serialization.py

```diff
@@ -210,6 +211,100 @@ def test_json_serialization():
 # ============================================================================
 
 
+def test_junit_serializer_initialization():
+    serializer = JunitSerializer(create_suite_result())
+
+
+def test_junit_serializer_initialization_with_incorrect_type_of_value():
+    assert_that(
+        calling(JunitSerializer).with_args(set()),
+        raises(
+            TypeError,
+            'Expected "SuiteResult" but got "set"')
+    )
+
+
+def check_junit_test_suite(test_suite):
+    assert_that(list(test_suite.attrib.keys()), ['errors', 'failures', 'name', 'tests', 'time', 'timestamp'])
+    assert_that(test_suite.tag, 'testsuite')
+
+
+def check_junit_test_case(test_case):
+    assert_that(list(test_case.attrib.keys()), ['classname', 'name', 'time'])
+    assert_that(test_case.tag, 'testcase')
+
+
+def test_junit_serialization():
+    suite_result = create_suite_result(
+        include_results_without_conditions=False,
+        include_results_without_display=False,
+        include_results_without_conditions_and_display=False
+    )
+    output = JunitSerializer(suite_result).serialize()
+
+    import xml.etree.ElementTree as ET
+
+    formatted_response = ET.fromstring(output)
+
+    assert_that(formatted_response.tag, 'testsuites')
+    assert_that(list(formatted_response.attrib.keys()), ['errors', 'failures', 'name', 'tests', 'time'])
+
+    for test_suite in list(formatted_response):
+        check_junit_test_case(test_suite)
+
+    for test_case in list(list(formatted_response)[0]):
+        check_junit_test_case(test_case)
+
+
+def test_junit_serialization_with_real_data():
```

Comment:
you can just put `iris_split_dataset_and_model` in the parameters, and it will be loaded from conftest.py